iips/IIP-0005/iip-0005.md

```diff
@@ -0,0 +1,164 @@
+discussions-to: https://github.com/iotaledger/IIPs/discussions/18
+status: Draft
+type: Standards Track
+layer: Interface
+created: 2025-07-22
+requires: None
+---
+
+## Abstract
+
+This proposal introduces a standardized interface for enabling ergonomic and application-specific queries to on-chain state. Move view functions are developer defined on-chain read API that can be easily queried off-chain without requiring transaction signing or state mutation (based on the Ethereum's [Solidity view function](https://docs.soliditylang.org/en/latest/contracts.html#view-functions) feature concept). The proposal is made to improve developer productivity and simplifying access to on-chain data through RPC, CLI, and SDK interfaces.
+
+
+
+## Motivation
+
+Currently, developers must write a significant amount of custom client-side logic to inspect on-chain data. This often involves multiple layers of fetching object IDs, deserializing raw bytes, and navigating dynamic fields just to reach an information stored within a Move object. While tools like [`dev-inspect`](https://docs.iota.org/iota-api-ref#iota_devinspecttransactionblock) or [`dry-run`](https://docs.iota.org/iota-api-ref#iota_dryruntransactionblock) exist, they suffer from limitations such as complexity in usage and difficulty in decoding return values.
+
+This proposal aims to address the following developer pain points:
+
+- Being forced to understand the logic of a Move data structure and then how to map that to RPC calls in order to fetch a piece of data, e.g., getting nested object ids.
+- Some data structures can't be accessed through current RPC calls because of the way they use dynamic fields, e.g., Bag and Table data structures.
+- Using `dev-inspect` can become complicated because the function return values the developer gets are [BCS](https://docs.iota.org/ts-sdk/bcs) bytes and then these need to be parsed, assuming the return type is known.
+
+## Specification
+
+This proposal defines a developer interface to support Move View Functions. The following is a list of specifications related to its implementation:
+
+- A View Function is a function in a Move module with a return type that does not alter the state of the ledger, i.e., when using the Move View Function interface no transactions are submitted to the network for their inclusion into the ledger.
+- Move View Functions are callable via at least one new RPC method that supports type parameters and function arguments.
+- The use of such interface SHOULD NOT require signature checks, i.e., invoking a Move View Function that takes as input an Owned Object SHOULD be made possible to anyone and not only to the owner of such object.
+- The use of such interface SHOULD NOT require the usage of a gas coin; spam attacks SHOULD be dealt with at the RPC level and not at the execution level; this is because the Move View Function does not alter the state of the ledger and thus cannot deduce gas.
```

Comment:
SHOULD NOT or MUST NOT? 

I would suggest to referring and comply to [RFC2119](https://www.rfc-editor.org/rfc/rfc2119) here.